### ISSUE TITLE
C Sample

### ISSUE THREAD

**Issue by sefai** [nektra/Deviare2#20]
Hello all,

Is there a C sample?

Regards,

**Comment by mxmauro** [comment]
Hi. Pure C sample no but there are C++ ones.

**Comment by sefai** [comment]
Thank you @mxmauro 

I am looking at the CTest sample. Command line parameters state that there is an option to add a Custom handler for each hook:

`  -ch=file: Custom handler to add to each hook (can be specified more than once).`

Is there a sample for this custom handler?

Regards,

**Comment by mxmauro** [comment]
Hi @sefai, yes you have RegistryPlugin sample. On C# you have Registry and FilePlugin. You can use them on CTest too.

**Comment by sefai** [comment]
@mxmauro I am sorry, but I couldn't understand what that plugin is doing.

I want to change the returned value (an _out parameter) from a kernel32 method call. How can I do this? Is there any guide or documentation that I can read so that I will stop distracting you?

Regards,

**Comment by sefai** [comment]
@mxmauro OK, I am making progress, I successfully changed the parameter. Now, I am trying to store some value on PreCall and use that value on PostCall. I see that there is WriteIntercallData and ReadIntercallData for that purpose, but I couldn't find any samples on how to use them. Can you help? 

PS: By the way, the from at address http://forum.nektra.com/forum/ seems to be down. I can only search on it using Google Cache. Are you guys aware of it?

Regards,

**Comment by bo3b** [comment]
Still Dead!  Google cache is now empty!  Terrific resource lost!

![image](https://user-images.githubusercontent.com/6544511/29240008-01d38748-7f11-11e7-9ec3-2b3f7adffbbb.png)

**Comment by Helios-vmg** [comment]
@bo3b I will look into it.

**Comment by bo3b** [comment]
Awesome, thank you for responding.  I was worried that you had abandoned Deviare.

**Comment by vlucky** [comment]
nektra forum is still has the issue [SQL ERROR as above mentioned]. could anybody fix the issue, please?

**Comment by vlucky** [comment]
Could anybody please reply?

**Comment by mxmauro** [comment]
Hi @vlucky,

You must provide a `VT_I#/VT_UI#/VT_INT/VT_UINT` or an array of `VT_I#/VT_UI#` (up to four items). You can create an array with `SafeArrayCreateVector`... i.e.:

```
VARIANT vt;

vt.vt = VT_ARRAY | VT_UINT;
vt.parray = SafeArrayCreateVector(VT_UINT, 0, 4)
```

And use SafeArray apis to store the values in the array before calling `CallCustomApi`

**Comment by vlucky** [comment]
Thanks @mxmauro ,

might be dumb question, but can I pass string too using above method?

**Comment by mxmauro** [comment]
No. Only integer values that the called function will receive. If you need to pass complex data, use VirtualAllocEx and WriteProcessMemory to write the data in the target process and pass the pointer as the parameter.
